packages/flutter/lib/src/widgets/radio_group.dart

```diff
@@ -0,0 +1,327 @@
+/// * Left and up arrow key: move selection to previous radio button in the
+///   group in reading order.
+///
+/// Arrow keys will wrap around if it reach the first or last radio in the
+/// group.
+///
+/// {@tool dartpad}
+/// Here is an example of RadioGroup widget.
+///
+/// Try using tab, arrow keys, and space to see how the widget responds.
+///
+/// ** See code in examples/api/lib/widgets/radio_group/radio_group.0.dart **
+/// {@end-tool}
+class RadioGroup<T> extends StatefulWidget {
+  /// creates a radio group
+  ///
+  /// The `groupValue` set the selection on a subtree radio with the same
+  /// [RawRadio.value].
+  ///
+  /// The `onChanged` is called when the selection has changed in the subtree
+  /// radios.
+  const RadioGroup({super.key, this.groupValue, required this.onChanged, required this.child})
+    : super();
+
+  /// The selected value under this radio group.
+  ///
+  /// [RawRadio] under this radio group where its [RawRadio.value] equals to this
+  /// value will be selected.
+  final T? groupValue;
+
+  /// Called when selection has changed.
+  ///
+  /// The value can be null if when unselect the [RawRadio] with
+  /// [RawRadio.toggleable] set to true.
+  final ValueChanged<T?> onChanged;
+
+  /// {@macro flutter.widgets.ProxyWidget.child}
+  final Widget child;
+
+  /// Gets the [RadioGroupRegistry] from the above the context.
+  ///
+  /// This registers a dependencies on the context that it causes rebuild
+  /// if [RadioGroupRegistry] has changed or its
+  /// [RadioGroupRegistry.groupValue] has changed.
+  static RadioGroupRegistry<T>? maybeOf<T>(BuildContext context) {
+    return context.dependOnInheritedWidgetOfExactType<_RadioGroupStateScope<T>>()?.state;
+  }
+
+  @override
+  State<StatefulWidget> createState() => _RadioGroupState<T>();
+}
+
+class _RadioGroupState<T> extends State<RadioGroup<T>> implements RadioGroupRegistry<T> {
+  late final Map<ShortcutActivator, Intent> _radioGroupShortcuts = <ShortcutActivator, Intent>{
+    const SingleActivator(LogicalKeyboardKey.arrowLeft): VoidCallbackIntent(_selectPreviousRadio),
+    const SingleActivator(LogicalKeyboardKey.arrowRight): VoidCallbackIntent(_selectNextRadio),
+    const SingleActivator(LogicalKeyboardKey.arrowDown): VoidCallbackIntent(_selectNextRadio),
+    const SingleActivator(LogicalKeyboardKey.arrowUp): VoidCallbackIntent(_selectPreviousRadio),
+    const SingleActivator(LogicalKeyboardKey.space): VoidCallbackIntent(_toggleFocusedRadio),
+  };
+
+  final Set<RadioClient<T>> _radios = <RadioClient<T>>{};
+
+  bool _debugCheckOnlySingleSelection() {
+    return _radios.where((RadioClient<T> radio) => radio.radioValue == groupValue).length < 2;
+  }
+
+  @override
+  T? get groupValue => widget.groupValue;
+
+  @override
+  void registerClient(RadioClient<T> radio) {
+    _radios.add(radio);
+    assert(
+      _debugCheckOnlySingleSelection(),
+      "RadioGroupPolicy can't be used for a radio group that allows multiple selection",
```

Comment:
Oh wait, this makes me wonder, if it is a multiple selection radio, does that mean users should use the deprecated APIs?

Reply:
Ah ok thanks!